packages/typescript-estree/src/create-program/createProjectProgram.ts

```diff
@@ -34,65 +34,82 @@ function createProjectProgram(
+    `ESLint was configured to run on \`${describedFilePath}\` using \`parserOptions.project\`: ${describedPrograms}`,
+  ];
+  let hasMatchedAnError = false;
 
-    const fileExtension = path.extname(extra.filePath);
-    if (!DEFAULT_EXTRA_FILE_EXTENSIONS.includes(fileExtension)) {
-      const nonStandardExt = `The extension for the file (${fileExtension}) is non-standard`;
-      if (extraFileExtensions.length > 0) {
-        if (!extraFileExtensions.includes(fileExtension)) {
-          errorLines.push(
-            `${nonStandardExt}. It should be added to your existing "parserOptions.extraFileExtensions".`,
-          );
-          hasMatchedAnError = true;
-        }
-      } else {
+  const extraFileExtensions = extra.extraFileExtensions || [];
+
+  extraFileExtensions.forEach(extraExtension => {
+    if (!extraExtension.startsWith('.')) {
+      errorLines.push(
+        `Found unexpected extension "${extraExtension}" specified with the "extraFileExtensions" option. Did you mean ".${extraExtension}"?`,
+      );
+    }
+    if (DEFAULT_EXTRA_FILE_EXTENSIONS.includes(extraExtension)) {
+      errorLines.push(
+        `You unnecessarily included the extension "${extraExtension}" with the "extraFileExtensions" option. This extension is already handled by the parser by default.`,
+      );
+    }
+  });
+
+  const fileExtension = path.extname(extra.filePath);
+  if (!DEFAULT_EXTRA_FILE_EXTENSIONS.includes(fileExtension)) {
+    const nonStandardExt = `The extension for the file (${fileExtension}) is non-standard`;
+    if (extraFileExtensions.length > 0) {
+      if (!extraFileExtensions.includes(fileExtension)) {
         errorLines.push(
-          `${nonStandardExt}. You should add "parserOptions.extraFileExtensions" to your config.`,
+          `${nonStandardExt}. It should be added to your existing "parserOptions.extraFileExtensions".`,
         );
         hasMatchedAnError = true;
       }
-    }
-
-    if (!hasMatchedAnError) {
+    } else {
       errorLines.push(
-        'The file must be included in at least one of the projects provided.',
+        `${nonStandardExt}. You should add "parserOptions.extraFileExtensions" to your config.`,
       );
+      hasMatchedAnError = true;
     }
+  }
 
-    throw new Error(errorLines.join('\n'));
+  if (!hasMatchedAnError) {
+    const [describedInclusions, describedSpecifiers] =
+      extra.projects.length === 1
+        ? ['that TSConfig does not', 'that TSConfig']
+        : ['none of those TSConfigs', 'one of those TSConfigs'];
+    errorLines.push(
+      `However, ${describedInclusions} include this file. Either:`,
+      `- Change ESLint's list of included files to not include this file`,
+      `- Change ${describedSpecifiers} to include this file`,
+      `- Create a new TSConfig that includes this file and include it in your parserOptions.project`,
+      `See the TypeScript ESLint docs for more info: https://typescript-eslint.io/docs/linting/troubleshooting#`,
```

Comment:
```suggestion
      `See the TypeScript ESLint docs for more info: https://typescript-eslint.io/docs/linting/troubleshooting/`,
```

or did you meant to hash link to a specific FAQ?

Reply:
Oh, right, it'd be good to link to the specific (newly updated) FAQ article.